rp2040-hal/src/pwm/reg.rs

```diff
@@ -0,0 +1,120 @@
+use super::dynslice::{DynSliceId, DynSliceMode};
+
+pub(super) unsafe trait RegisterInterface {
+    /// Provide a [`DynSliceId`] identifying the set of registers controlled by
+    /// this type.
+    fn id(&self) -> DynSliceId;
+
+    #[inline]
+    fn advance_phase(&self) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.csr.modify(|_, w| w.ph_adv().set_bit())
+    }
+
+    #[inline]
+    fn retard_phase(&self) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.csr.modify(|_, w| w.ph_ret().set_bit())
+    }
+
+    #[inline]
+    fn do_change_mode(&self, mode: DynSliceMode) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        match mode {
+            DynSliceMode::FreeRunning => ch.csr.modify(|_, w| w.divmode().div()),
+            DynSliceMode::InputHighRunning => ch.csr.modify(|_, w| w.divmode().level()),
+            DynSliceMode::CountRisingEdge => ch.csr.modify(|_, w| w.divmode().rise()),
+            DynSliceMode::CountFallingEdge => ch.csr.modify(|_, w| w.divmode().fall()),
+        }
+    }
+
+    #[inline]
+    fn write_inv_a(&mut self, value: bool) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.csr.write(|w| w.a_inv().bit(value));
+    }
+
+    #[inline]
+    fn write_inv_b(&mut self, value: bool) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.csr.write(|w| w.b_inv().bit(value));
+    }
+
+    #[inline]
+    fn write_ph_correct(&mut self, value: bool) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.csr.write(|w| w.ph_correct().bit(value));
+    }
+
+    #[inline]
+    fn enable(&self, value: bool) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.csr.modify(|_, w| w.en().bit(value));
+    }
+
+    #[inline]
+    fn write_div_int(&mut self, value: u8) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.div.write(|w| unsafe { w.int().bits(value) });
+    }
+    #[inline]
+    fn write_div_frac(&mut self, value: u8) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.div.write(|w| unsafe { w.frac().bits(value) });
+    }
+
+    #[inline]
+    fn write_ctr(&mut self, value: u16) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.ctr.write(|w| unsafe { w.ctr().bits(value) });
+    }
+
+    #[inline]
+    fn write_cc_a(&mut self, value: u16) {
+        let num = self.id().num as usize;
+        let ch = unsafe { &(*pac::PWM::ptr()).ch[num] };
+        ch.cc.write(|w| unsafe { w.a().bits(value) });
```

Comment:
This overwrites the value written to channel b. Use
```suggestion
        ch.cc.modify(|_, w| unsafe { w.a().bits(value) });
```
instead